compiler+runtime/src/cpp/clojure/core_native.cpp

```diff
@@ -702,6 +702,27 @@ jank_object_ptr jank_load_clojure_core_native()
     intern_fn_obj("range", fn);
   }
 
+  {
+    auto const fn(
+      make_box<obj::jit_function>(behavior::callable::build_arity_flags(3, false, false)));
+    fn->arity_0 = []() -> object * {
+      return iterate(__rt_ctx->intern_var("clojure.core", "inc").expect_ok()->deref(), make_box(0));
+    };
+    fn->arity_1 = [](object * const end) -> object * {
+      return obj::integer_range::create(expect_object<obj::integer>(end)->data);
+    };
+    fn->arity_2 = [](object * const start, object * const end) -> object * {
+      return obj::integer_range::create(expect_object<obj::integer>(start)->data,
+                                        expect_object<obj::integer>(end)->data);
+    };
+    fn->arity_3 = [](object * const start, object * const end, object * const step) -> object * {
+      return obj::integer_range::create(expect_object<obj::integer>(start)->data,
+                                        expect_object<obj::integer>(end)->data,
+                                        expect_object<obj::integer>(step)->data);
```

Comment:
You're using `expect_object` here. Are you 100% sure that these fns can never be called with anything other than an integer for each arg? If they are, `expect_object` will not check and we'll have undefined behavior. At best, it will crash. At worst, it will launch missiles.

Reply:
These fns should only have integer args because for an integer-range to be called it uses the integer? function to make sure the function arguments are all integers.

Reply:
You're exposing a public var here: `clojure.core-native/integer-range`

Sure, in the one place you're calling it, all of the inputs are boxed integers. But let's talk about good practices. Is this the assumption we want to make for all potential callers of this function? What're the pros/cons of the choice?